presto-hudi/src/main/java/com/facebook/presto/hudi/HudiModule.java

```diff
@@ -132,6 +138,34 @@ public ExecutorService createCachingHiveMetastoreExecutor(MetastoreClientConfig
                 daemonThreadsNamed("hive-metastore-hudi-%s"));
     }
 
+    @ForHudiSplitAsyncQueue
+    @Singleton
+    @Provides
+    public ExecutorService createHudiSplitManagerExecutor()
+    {
+        return newCachedThreadPool(daemonThreadsNamed("hudi-split-manager-%s"));
+    }
+
+    @ForHudiSplitSource
+    @Singleton
+    @Provides
+    public ScheduledExecutorService createSplitLoaderExecutor(HudiConfig hudiConfig)
+    {
+        return newScheduledThreadPool(
+                hudiConfig.getSplitLoaderParallelism(),
+                daemonThreadsNamed("hudi-split-loader-%s"));
+    }
+
+    @ForHudiBackgroundSplitLoader
+    @Singleton
+    @Provides
+    public ExecutorService createSplitGeneratorExecutor(HudiConfig hudiConfig)
+    {
```

Comment:
same. asking since these are not cached thread pools.

Reply:
This one is a fixed thread pool as it is IO bound, mainly used for file listing. If we use cached thread pool, then number of threads can keep on increasing as the number of partition increases. In the case concurrent queries running on a server, some queries can hog all the resources if we use cached thread pool. 
FWIW, I tested both on tpcds data with ~2k partitions. It did not make much difference and the queries were not running concurrently. But, I think if the cluster is shared by several clients and partitions are of the order of ~10k, cached thread pool might block a few queries.